pageserver/src/tenant.rs

```diff
@@ -1461,51 +1622,45 @@ impl Tenant {
             // For configurations without remote storage, we tolerate that we're not crash-safe here.
             // The timeline may come up Active but with missing layer files, in such setups.
             // See https://github.com/neondatabase/neon/pull/3919#issuecomment-1531726720
-            std::fs::remove_dir_all(&local_timeline_directory).with_context(|| {
-                format!(
-                    "Failed to remove local timeline directory '{}'",
-                    local_timeline_directory.display()
-                )
-            })?;
+            if let Err(e) = std::fs::remove_dir_all(local_timeline_directory) {
+                warn!("failed to remove timeline files after uploading tombstoned index_part.json: {e}");
+            }
```

Comment:
Unacceptable to treat failure to remove the files as a success case.
This is completely retryable.

See how I did it in https://github.com/neondatabase/neon/pull/4147

Reply:
In which cases is it actually retryable? To support directory permission changes?

In general, I guess it could be done, I just failed to think of the retryable cases. No problem adding new variants.

Your pr could most likely target this branch though I haven't looked at after yesterday.

Reply:
The problem is that we would call the delete_timeline operation as succeeded, but, there's actually still state on the local pageserver.
IMO that is an unacceptable outcome.